gtsam/geometry/Line3.h

```diff
@@ -0,0 +1,178 @@
+            R_(R), a_(a), b_(b)
+    {}
+
+    /** Sample two points lying on the line
+     * @return Vector6 where first 3 elements are one point, and
+     * next 3 elements are another point **/
+    Vector6 points();
+
+    /**
+     * The retract method maps from the tangent space back to the manifold.
+     * The tangent space for the rotation of a line is only two dimensional -
+     * rotation about x and y
+     * @param v: increment in tangent space
+     * @param H: Jacobian of retraction with respect to the increment
+     * @return: resulting line after adding the increment and mapping to the manifold
+     */
+    Line3 retract(const Vector4 &v, OptionalJacobian<4,4> H = boost::none) const;
+
+    /**
+     * The localCoordinates method is the inverse of retract and finds the difference
+     * between two lines in the tangent space.
+     * @param q Line3 on manifold
+     * @param H OptionalJacobian of localCoordinates with respect to line
+     * @return Vector4 difference in the tangent space
+     */
+    Vector4 localCoordinates(const Line3 &q, OptionalJacobian<4,4> H = boost::none) const;
+
+    /**
+     * Rotation of line accessor
+     * @return Rot3
+     */
+    Rot3 R() const
+    {
+        return R_;
+    }
+
+    /**
+     * Accessor for a, b
+     * @return Vector2(a, b)
+     */
+    Vector2 V() const
+    {
+        return Vector2(a_, b_);
+    }
+
+    /**
+     * Print R, a, b
+     * @param s: optional starting string
+     */
+    void print(const std::string &s = "") const;
+
+    /**
+     * Check if two lines are equal
+     * @param l2 - line to be compared
+     * @param tol : optional tolerance
+     * @return boolean - true if lines are equal
+     */
+    bool equals(const Line3 &l2, double tol = 10e-9) const;
+
+    /**
+     * Projecting a line to an image plane
+     * @param Dline: OptionalJacobian of projected line with respect to this line
+     * @return Point3 - projected line in image plane
+     */
+    Point3 project(OptionalJacobian<3, 4> Dline = boost::none) const;
+};
+
+template <>
+struct traits<Line3> : public internal::Manifold<Line3> {};
+
+template <>
+struct traits<const Line3> : public internal::Manifold<Line3> {};
+
+// For expression factor graphs
+typedef Expression<Line3> Line3_;
+
+/**
+ * Function projects input line to image plane
+ * @param L input line
+ * @param Dline OptionalJacbian of projected line with respect to input line
+ * @return Point3 - projected line
+ */
+Point3 projectLine(const Line3 &L,
+        OptionalJacobian<3, 4> Dline = boost::none);
+
+/**
+ * Transform a line from world to camera frame
+ * @param p - Pose3 of camera in world frame
+ * @param l - Line3 in world frame
+ * @param Dpose - OptionalJacobian of transformed line with respect to p
+ * @param Dline -  OptionalJacobian of transformed line with respect to l
+ * @return Transformed line in camera frame
+ */
+Line3 transformTo(const Pose3 &p, const Line3 &l,
```

Comment:
Can you use dTs naming convention for all poses T and rotation matrices R? Where d = destination and s = source, such that dPoint = dTs * sPoint.